### PR TITLE
feat: introduce helm-docs for consistency

### PR DESCRIPTION
**Pull request by oliverbaehler**
Signed-off-by: Oliver Bähler <oliverbaehler@hotmail.com>

Hi Guys, With this PR i added `helm-docs` for consistent value documentation. If you don't like the change, you can close the pr :).

Makefile:
  * Added `make helm-docs` to generate doc locally
  * Added `make helm-lint` to lint chart locally

Workflow:
  * Added diff to verify documentation is consistent with values

Helm Values:
  * Added `podLabels`
  * Added `priorityClassName`

### CODE REVIEW

**Comment by chanwit** [comment]
Thank you so much @oliverbaehler 
I'll take a look at it shortly!

**Comment by tomhuang12** [comment]
Love this! Thank you 🏅 . Just a minor nit to bump the chart version to pass the helm lint.

**Comment by oliverbaehler** [comment]
@tomhuang12 thanks, rebase everything. @tomhuang12 i also added that the tag is used from `$.Chart.AppVersion` so you don't have to bump the tag versions in the values.yaml for both the controller and runner image everytime

File: charts/tf-controller/templates/deployment.yaml [hunk 1]
```diff
@@ -46,12 +49,12 @@ spec:
             fieldRef:
               fieldPath: metadata.namespace
         - name: RUNNER_POD_IMAGE
-          value: "{{ .Values.runner.image.repository }}:{{ .Values.runner.image.tag}}"
+          value: "{{ .Values.runner.image.repository }}:{{ default $.Chart.AppVersion .Values.runner.image.tag }}"
```
Comment (tomhuang12) [inline charts/tf-controller/templates/deployment.yaml:52]: I don't think you need `$` as it is not in a particular scope and should be able to access `.Chart` without it.

File: charts/tf-controller/templates/deployment.yaml [hunk 2]
```diff
@@ -46,12 +49,12 @@ spec:
...
         {{- range $key, $value := .Values.extraEnv }}
         - name: {{ $key }}
           value: {{ $value }}
         {{- end }}
-        image: "{{ .Values.image.repository }}:{{ .Values.image.tag}}"
+        image: "{{ .Values.image.repository }}:{{ default $.Chart.AppVersion $.Values.image.tag }}"
```
Comment (tomhuang12) [inline charts/tf-controller/templates/deployment.yaml:57]: Same here

**Review by tomhuang12 — COMMENTED** [review]
Thank you! Didn't think about that one. Just minor nit.

**Comment by chanwit** [comment]
We keep releasing, and the chart version causes an unnecessary error. 
I think I shall merge this PR, and we can continue from there.

**Review by chanwit — APPROVED** [review]
LGTM
thank you @oliverbaehler :1st_place_medal:

**Pull request merged**